lang/cs/Org.Apache.REEF.Network/Elastic/Topology/Physical/Default/OneToNTopology.cs

```diff
@@ -0,0 +1,269 @@
+        /// <param name="message">Message from the driver</param>
+        public override void OnNext(DriverMessagePayload message)
+        {
+            switch (message.PayloadType)
+            {
+                case DriverMessagePayloadType.Failure:
+                    {
+                        var rmsg = message as TopologyMessagePayload;
+
+                        foreach (var updates in rmsg.TopologyUpdates)
+                        {
+                            foreach (var node in updates.Children)
+                            {
+                                Log.Log(Level.Info, "Removing task {0} from the topology.", node);
+                                _nodesToRemove.TryAdd(node, new byte());
+                                _commLayer.RemoveConnection(node);
+                            }
+                        }
+                        break;
+                    }
+                case DriverMessagePayloadType.Update:
+                    {
+                        if (_sendQueue.Count > 0)
+                        {
+                            if (_sendQueue.TryPeek(out ElasticGroupCommunicationMessage toSendmsg))
+                            {
+                                var rmsg = message as TopologyMessagePayload;
+
+                                if (_piggybackTopologyUpdates)
+                                {
+                                    var toSendmsgWithTop = toSendmsg as DataMessageWithTopology;
+                                    var updates = rmsg.TopologyUpdates;
+
+                                    UpdateTopology(ref updates);
+                                    toSendmsgWithTop.TopologyUpdates = updates;
+                                }
+
+                                foreach (var taskId in _nodesToRemove.Keys)
+                                {
+                                    var id = Utils.GetTaskNum(taskId);
+                                    _nodesToRemove.TryRemove(taskId, out byte val);
+                                    _children.TryRemove(id, out string str);
+                                }
+                            }
+
+                            // Unblock this broadcast round.
+                            _topologyUpdateReceived.Set();
+                        }
+                        else
+                        {
+                            Log.Log(Level.Warning, "Received a topology update message from driver "
+                                + "but sending queue is empty: ignoring.");
+                        }
+                    }
+                    break;
+
+                default:
+                    throw new ArgumentException(
+                        $"Message type {message.PayloadType} not supported by N to one topologies.");
+            }
+        }
+
+        private void UpdateTopology(ref List<TopologyUpdate> updates)
+        {
+            TopologyUpdate toRemove = null;
+            foreach (var update in updates)
```

Comment:
maybe
```C#
var update = updates.Find(elem => elem.Node == TaskId);

if (update != null)
{
    foreach (var child in update.Children)
    {
        if (!_nodesToRemove.TryRemove(child, out byte value))
        {
            var id = Utils.GetTaskNum(child);
            _children.TryAdd(id, child);
        }
    }

    updates.Remove(update);
}
```

Reply:
I haved added `update != null` because `Find` will return `null` if no update for a node is found.

Reply:
I believe you have to move the whole thing into `if (update != null)` block - otherwise you'll get an NPE in `foreach (var child in update.Children)` (because `update` can be `null`)

Reply:
of course...Fixed.